src/helpers.js

```diff
@@ -35,6 +39,37 @@ function decodeEventBody(event) {
   return event;
 }
 
+function decodeToken(event) {
+  const existingToken = event.cookieObject[apiTokenName()];
+
+  try {
+    event.userToken = new ApiToken(existingToken);
+  } catch (error) {
+    event.userToken = new ApiToken();
+  }
+  if (isFromReadingRoom(event)) {
+    event.userToken.readingRoom();
+  }
+
+  // do I have to worry about removing a reading room flag from existing token?
+  return event;
+}
+
+function encodeToken(event, response) {
```

Comment:
Isn't this where it should just return the existing response if `! event._userTokenUpdated`? I don't hate this, but I think it'd be best to avoid sending back a `Set-Cookie` when we don't need to if we can do so reliably.